cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java

```diff
@@ -66,15 +68,17 @@
   public void datasetLineage(HttpRequest request, HttpResponder responder,
                              @PathParam("namespace-id") String namespaceId,
                              @PathParam("dataset-id") String datasetId,
-                             @QueryParam("start") @DefaultValue("-1") long start,
-                             @QueryParam("end") @DefaultValue("-1") long end,
+                             @QueryParam("start") String startStr,
+                             @QueryParam("end") String endStr,
                              @QueryParam("levels") @DefaultValue("10") int levels) throws Exception {
 
-    checkArguments(start, end, levels);
+    checkLevels(levels);
+    TimeRange range = parseRange(startStr, endStr);
 
     Id.DatasetInstance datasetInstance = Id.DatasetInstance.from(namespaceId, datasetId);
-    Lineage lineage = lineageGenerator.computeLineage(datasetInstance, start, end, levels);
-    responder.sendJson(HttpResponseStatus.OK, new LineageRecord(start, end, lineage.getRelations()),
+    Lineage lineage = lineageGenerator.computeLineage(datasetInstance, range.getStart(), range.getEnd(), levels);
+    responder.sendJson(HttpResponseStatus.OK,
+                       new LineageRecord(range.getStart(), range.getEnd(), lineage.getRelations()),
```

Comment:
When we change to take in seconds from user, we need to change what we return also into seconds for consistency.